### ISSUE TITLE
Ubuntu 24.04 LTS Noble Numbat

### ISSUE THREAD

**Issue by vsarunas** [apple/swift-docker#382]
It would be highly advantageous for the Swift 6 development process to consider building and releasing Ubuntu 24.04 LTS packages. This new release of Ubuntu, detailed [here](https://canonical.com/blog/canonical-releases-ubuntu-24-04-noble-numbat) and further elaborated in the [release notes](https://discourse.ubuntu.com/t/noble-numbat-release-notes/39890), offers extended support that lasts for five years, until June 2029.

Currently, Swift packages are available for Ubuntu 18.04, but this older version reached the end of its standard support in May 2023, as noted [here](https://ubuntu.com/blog/18-04-end-of-standard-support).

**Comment by finagolfin** [comment]
A docker image for 24.04 was added last month, #380, that is the first step towards producing builds.

**Comment by vsarunas** [comment]
Thanks!

I had looked in https://github.com/apple/swift-docker/tree/main/nightly-6.0/ubuntu prior creating the case rather than the master folder.

**Comment by finagolfin** [comment]
@shahmishal, any plans to add an official 24.04 build for the next 5.10 patch release? I noticed that the github CI is preparing a runner image for 24.04 and left Swift out, actions/runner-images#9848, likely because there is no official 24.04 release of 5.10.

**Comment by hassila** [comment]
https://forums.swift.org/t/announcing-swift-5-10-1-for-linux-and-windows/72292

**Comment by finagolfin** [comment]
Yes, I saw and [already notified the github people](https://github.com/actions/runner-images/issues/9848#issuecomment-2152881554), but technically this issue is about the upcoming Swift 6 supporting 24.04, which hasn't been added yet, so this issue can be kept open if the original reporter wants.

**Comment by FranzBusch** [comment]
@shahmishal We already have Swift release images for Ubuntu noble can we also get nightly 6 and main noble images?

**Comment by finagolfin** [comment]
[It's being worked on](https://forums.swift.org/t/where-are-the-rest-of-the-ubuntu-24-04-toolchains/74180/2).

**Comment by xtremekforever** [comment]
Any news on this? Have the issues with Swift on Noble been fixed up? I'm creating my own containers that I can use for now, but would be nice to get official ones.

**Comment by shahmishal** [comment]
Ubuntu 24.04 images are available https://www.swift.org/install/linux/ubuntu/24_04/#latest

**Comment by xtremekforever** [comment]
@shahmishal ah no I meant for generating the nightly containers for Noble as well. The [main](https://download.swift.org/development/ubuntu2404/swift-DEVELOPMENT-SNAPSHOT-2024-12-12-a/swift-DEVELOPMENT-SNAPSHOT-2024-12-12-a-ubuntu24.04.tar.gz) and [6.1](https://download.swift.org/swift-6.1-branch/ubuntu2404/swift-6.1-DEVELOPMENT-SNAPSHOT-2024-12-12-a/swift-6.1-DEVELOPMENT-SNAPSHOT-2024-12-12-a-ubuntu24.04.tar.gz) toolchains are already being generated for Ubuntu 24.04, so how hard would it be to just add a ubuntu24.04 container that is generated with this toolchain installed?

- https://github.com/swiftlang/swift-docker/tree/main/nightly-main/ubuntu
- https://github.com/swiftlang/swift-docker/tree/main/nightly-6.1/ubuntu

**Comment by vsarunas** [comment]
Preparing for Swift 6.1 release - noticed missing nightly images for Ubuntu 24.04 (Noble). 

I hope this PR minimises the work required to have this published: https://github.com/swiftlang/swift-docker/pull/438

**Comment by xtremekforever** [comment]
We now have containers for Ubuntu 24.04 nightlies!

<img width="743" alt="Image" src="https://github.com/user-attachments/assets/2f7d5e27-29c4-4d42-bc28-8d928bd45e7d" />

I think this ticket can be closed at this point :)

**Comment by xtremekforever** [comment]
Since this is done, can you close the issue @vsarunas ?
